src/sys/unix.rs

```diff
@@ -1506,6 +1511,22 @@ impl crate::Socket {
         }
     }
 
+    /// Returns list of CCIDs supported by the endpoint
+    #[cfg(target_os = "linux")]
+    pub fn dccp_available_ccids(&self) -> io::Result<Vec<u8>> {
+        let mut buf: [MaybeUninit<u8>; 10] = unsafe { MaybeUninit::uninit().assume_init() };
+        let mut len = buf.len() as libc::socklen_t;
+        syscall!(getsockopt(
+            self.as_raw(),
+            libc::SOL_DCCP,
+            libc::DCCP_SOCKOPT_AVAILABLE_CCIDS,
+            buf.as_mut_ptr().cast(),
+            &mut len,
+        ))?;
+        let buf = &buf[..len as usize - 1];
+        Ok(unsafe { &*(buf as *const [_] as *const [u8]) }.into())
```

Comment:
Can we do better than allocating 10 bytes? Maybe a small vec like type?

Reply:
I'm not quiet sure what to do here. I've changed the buf to length 5 instead of 10. The code of this function was basically copied from the device function.
https://github.com/rust-lang/socket2/blob/4c436818cddec880c26c4e79aca7106ba88827ff/src/sys/unix.rs#L1488-L1507